### PR TITLE
Add Tween.timeScale(scale) method and Tween.duration getter

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #16973

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/tween/tween.ts [hunk 1]
```diff
@@ -499,7 +539,9 @@ export class Tween<T extends object = any> {
             action = actions.pop();
         }
 
-        if (action instanceof ActionInterval) {
+        if (action && actions.length !== 0) {
+            actions.push(repeat(action, Number.MAX_SAFE_INTEGER));
```
Comment (dumganhar) [inline cocos/tween/tween.ts:546]: Also fix the bug of repeatForever. Reported at: https://forum.cocos.org/t/topic/152139

File: cocos/tween/tween.ts [hunk 2]
```diff
@@ -101,7 +102,10 @@ export class Tween<T extends object = any> {
      */
     then<U extends object = any> (other: Tween<U>): Tween<T> {
         const u = other._union(true);
-        if (u) this._actions.push(u);
+        if (u) {
+            u.setSpeed(other._timeScale);
```
Comment (minggo) [inline cocos/tween/tween.ts:106]: Why pass time scale to speed?

File: cocos/tween/actions/action.ts [hunk 3]
```diff
@@ -242,8 +242,11 @@ export abstract class Action {
  * @extends Action
  */
 export abstract class FiniteTimeAction extends Action {
-    _duration = 0;
-    _timesForRepeat = 1;
+    protected _duration = 0;
+
+    getDurationScaled (): number {
```
Comment (minggo) [inline cocos/tween/actions/action.ts:247]: What's the difference between time scale and duration scale?

File: cocos/tween/tween.ts (hunk 2 above)
Comment (dumganhar) [reply cocos/tween/tween.ts:106]: Action.setSpeed is the old method in Action class. Just use it.
Yep, if needed, we could rename Action.setSpeed to Action.setTimeScale.

File: cocos/tween/actions/action.ts (hunk 3 above)
Comment (dumganhar) [reply cocos/tween/actions/action.ts:247]: If an action is an ActionInterval instance, `getDurationScaled` will return `this._duration * this._timeScale`.

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Pull request merged**